File: design-patterns/java/Strategy-DesignPattern/WithoutStrategyPattern/PaymentSystem.java
package WithoutStrategyPattern;

// The "Bad" Way: All logic in one place
public class PaymentSystem {
    public void processPayment(String type, int amount) {
        if (type.equalsIgnoreCase("CREDIT_CARD")) {
            System.out.println("Processing Credit Card payment of $" + amount);
        } else if (type.equalsIgnoreCase("PAYPAL")) {
            System.out.println("Processing PayPal payment of $" + amount);
        } else if (type.equalsIgnoreCase("BANK_TRANSFER")) {
            System.out.println("Processing Bank Transfer payment of $" + amount);
        } else if (type.equalsIgnoreCase("Gpay")) {
            System.out.println("Processing Bank Transfer payment of $" + amount);
        } else {
            System.out.println("Payment method not supported.");
        }
    }
}

/*
  The Problem:
  If you want to add "Google Pay," you must open this file
  and add another else if. This is risky and messy.
 */

File: design-patterns/python/Strategy-DesignPattern/WithoutStrategyPattern/payment_system.py
"""Without strategy: all logic in one place (if/else)."""


class PaymentSystem:
    def process_payment(self, payment_type: str, amount: int) -> None:
        t = payment_type.upper()
        if t == "CREDIT_CARD":
            print(f"Processing Credit Card payment of ${amount}")
        elif t == "PAYPAL":
            print(f"Processing PayPal payment of ${amount}")
        elif t == "BANK_TRANSFER":
            print(f"Processing Bank Transfer payment of ${amount}")
        elif t == "GPAY":
            print(f"Processing Gpay payment of ${amount}")
        else:
            print("Payment method not supported.")
